### PR TITLE
add more information in Error Message of httpproxy

### LINKED ISSUE #5638: add more information in Error Message of httpproxy
Root httpproxy will be invalid if child proxies are not config correctly, but with current error message, it is difficult to figure out which child proxy has the problem, could you please include more useful information? for example, it will be very helpful to add namespace or child proxy name in the Message.

    Errors:
      Message:             root httpproxy cannot include another root httpproxy
      Reason:              RootIncludesRoot
      Status:              True
      Type:                IncludeError

### PR DESCRIPTION
**Pull request by HeavenTonight**
Fixes #5638 
add more information in Error Message of httpproxy

### CODE REVIEW

**Comment by skriss** [comment]
There are also some failing unit tests that need to be looked into.

File: internal/dag/httpproxy_processor.go [hunk 1]
```diff
@@ -715,7 +715,7 @@ func (p *HTTPProxyProcessor) computeRoutes(
 
 		if includedProxy.Spec.VirtualHost != nil {
 			validCond.AddErrorf(contour_api_v1.ConditionTypeIncludeError, "RootIncludesRoot",
-				"root httpproxy cannot include another root httpproxy")
+				"root httpproxy cannot include another root httpproxy,child httpproxy info is [namespace:%s name:%s]", includedProxy.Namespace, includedProxy.Name)
```
[the remark this answers is not in the document]
Comment (sunjayBhatia) [reply internal/dag/httpproxy_processor.go:718]: yep that works for me 👍🏽 concise and effective

File: internal/dag/status_test.go [hunk 2]
```diff
@@ -1052,7 +1053,7 @@ func TestDAGStatus(t *testing.T) {
 		want: map[types.NamespacedName]contour_api_v1.DetailedCondition{
 			{Name: proxyIncludesRootDifferentFQDN.Name, Namespace: proxyIncludesRootDifferentFQDN.Namespace}: fixture.NewValidCondition().
 				WithGeneration(proxyIncludesRootDifferentFQDN.Generation).
-				WithError(contour_api_v1.ConditionTypeIncludeError, "RootIncludesRoot", "root httpproxy cannot include another root httpproxy"),
+				WithError(contour_api_v1.ConditionTypeIncludeError, "RootIncludesRoot", fmt.Sprintf("root httpproxy cannot include another root httpproxy (%s/%s)", proxyIncludesRootDifferentFQDN.Namespace, proxyIncludesRootDifferentFQDN.Name)),
```
Comment (skriss) [inline internal/dag/status_test.go:1056]: The namespace/name here should be of the child proxy, not the parent:
```suggestion
				WithError(contour_api_v1.ConditionTypeIncludeError, "RootIncludesRoot", fmt.Sprintf("root httpproxy cannot include another root httpproxy (%s/%s)", proxyRootIncludedByRootDiffFQDN.Namespace, proxyRootIncludedByRootDiffFQDN.Name)),
```

File: internal/dag/status_test.go [hunk 3]
```diff
@@ -1625,7 +1626,7 @@ func TestDAGStatus(t *testing.T) {
 		objs: []any{proxyTCPIncludesFoo, proxyValidTCPRoot, fixture.ServiceRootsKuard},
 		want: map[types.NamespacedName]contour_api_v1.DetailedCondition{
 			{Name: proxyTCPIncludesFoo.Name, Namespace: proxyTCPIncludesFoo.Namespace}: fixture.NewValidCondition().
-				WithError(contour_api_v1.ConditionTypeTCPProxyIncludeError, "RootIncludesRoot", "root httpproxy cannot include another root httpproxy"),
+				WithError(contour_api_v1.ConditionTypeTCPProxyIncludeError, "RootIncludesRoot", fmt.Sprintf("root httpproxy cannot include another root httpproxy (%s/%s)", proxyTCPIncludesFoo.Namespace, proxyTCPIncludesFoo.Name)),
```
Comment (skriss) [inline internal/dag/status_test.go:1629]: The namespace/name here should be of the child proxy, not the parent:
```suggestion
				WithError(contour_api_v1.ConditionTypeTCPProxyIncludeError, "RootIncludesRoot", fmt.Sprintf("root httpproxy cannot include another root httpproxy (%s/%s)", proxyValidTCPRoot.Namespace, proxyValidTCPRoot.Name)),
```

**Review by skriss — CHANGES_REQUESTED** [review]
There are still some failing tests to fix per below comments. Also, please add a changelog file by following the instructions at https://github.com/projectcontour/contour/actions/runs/5949094558/job/16145697974?pr=5670

**Comment by HeavenTonight** [comment]
> There are still some failing tests to fix per below comments. Also, please add a changelog file by following the instructions at https://github.com/projectcontour/contour/actions/runs/5949094558/job/16145697974?pr=5670
> 
> 根据以下评论，仍有一些失败的测试需要修复。 另外，请按照 https://github.com/projectcontour/contour/actions/runs/5949094558/job/16145697974?pr=5670 中的说明添加更改日志文件

thinks for review,i will fix it soon !

File: changelogs/unreleased/5670-HeavenTonight-small.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Add namespace or child proxy name in the message when Root httpproxy will be invalid if child proxies are not config correctly.
```
Comment (sunjayBhatia) [inline changelogs/unreleased/5670-HeavenTonight-small.md:1]:
```suggestion
Add included HTTPProxy namespace and name to status condition error message when a root HTTPProxy includes another root.
```

**Review by sunjayBhatia — CHANGES_REQUESTED** [review]

File: changelogs/unreleased/5670-HeavenTonight-small.md (hunk 4 above)
Comment (HeavenTonight) [reply changelogs/unreleased/5670-HeavenTonight-small.md]: Thanks for review 😉

**Review by skriss — APPROVED** [review]
LGTM, thanks @HeavenTonight

**Pull request merged**